app/helpers/shared_helpers.py

```diff
@@ -1,3 +1,31 @@
+try:
+    from urlparse import urlparse, parse_qs
+except ImportError:
+    from urllib.parse import urlparse, parse_qs
+
+
+def parse_links(links):
```

Comment:
Given that we only ever use each link separately this function could be rewritten as
```
def parse_link(links, label):
  return parse_qs(urlparse(links[label]).query) if label in links else None
```